presto-hive/src/test/java/com/facebook/presto/hive/TestHiveRecoverableGroupedExecution.java

```diff
@@ -332,6 +347,28 @@ public void testUnionAll(int writerConcurrency)
                         "DROP TABLE IF EXISTS test_failure"));
     }
 
+    @Test
+    public void testCountOnUnbucketedTable()
+            throws Exception
+    {
+        queryRunner = createQueryRunner(true);
+        testRecoverableGroupedExecution(
+                4,
+                ImmutableList.of(
+                        "CREATE TABLE test_table AS\n" +
+                                "SELECT orderkey, comment\n" +
+                                "FROM orders\n"),
+                "CREATE TABLE test_success AS\n" +
+                        "SELECT count(*) as a, comment FROM test_table group by comment",
```

Comment:
One question is how to do make sure the retry credits to stage retry (instead of bucket retry?)  -- for example, if the shuffle is already finished then kill a random worker will enable bucket retry right? 

One idea is to turn off bucket retry in this test?